spring-security/src/test/java/com/sap/cloud/security/spring/config/XsuaaServiceConfigurationTest.java

```diff
@@ -5,26 +5,54 @@
  */
 package com.sap.cloud.security.spring.config;
 
+import com.sap.cloud.environment.servicebinding.SapVcapServicesServiceBindingAccessor;
+import com.sap.cloud.security.config.OAuth2ServiceConfiguration;
+import com.sap.cloud.security.config.ServiceBindingEnvironment;
+import com.sap.cloud.security.config.ServiceConstants;
+import org.apache.commons.io.IOUtils;
+import org.junit.jupiter.api.BeforeAll;
 import org.junit.jupiter.api.Test;
+import org.springframework.beans.factory.annotation.Autowired;
 import org.springframework.boot.context.properties.EnableConfigurationProperties;
-import org.springframework.boot.test.context.runner.ApplicationContextRunner;
+import org.springframework.boot.test.context.SpringBootTest;
+import org.springframework.context.annotation.Configuration;
+import org.springframework.context.annotation.PropertySource;
 
+import java.io.IOException;
+
+import static java.nio.charset.StandardCharsets.UTF_8;
 import static org.junit.jupiter.api.Assertions.assertEquals;
 
+
+@SpringBootTest(classes = { SingleXsuaaConfigurationFromFile.class })
 class XsuaaServiceConfigurationTest {
 
-	private final ApplicationContextRunner runner = new ApplicationContextRunner();
+	static ServiceBindingEnvironment env;
+
+	@Autowired
+	XsuaaServiceConfiguration xsuaaConfig;
 
-	@EnableConfigurationProperties(XsuaaServiceConfiguration.class)
-	static class EnablePropertiesConfiguration {
+	@BeforeAll
+	static void setup() throws IOException {
+		String serviceBindingJson = IOUtils.resourceToString("/singleXsuaaAndIasBinding.json", UTF_8);
+		env = new ServiceBindingEnvironment(new SapVcapServicesServiceBindingAccessor(any -> serviceBindingJson));
 	}
 
 	@Test
 	void configuresXsuaaServiceConfiguration() {
-		runner.withUserConfiguration(EnablePropertiesConfiguration.class)
-				.withPropertyValues("sap.security.services.xsuaa.url:http://localhost",
-						"sap.security.services.xsuaa.uaadomain:localhost", "sap.security.services.xsuaa.clientid:cid")
-				.run(context -> assertEquals("http://localhost",
-						context.getBean(XsuaaServiceConfiguration.class).getUrl().toString()));
+		assertConfigsAreEqual(xsuaaConfig, env.getXsuaaConfiguration());
+    }
+
+	static void assertConfigsAreEqual(XsuaaServiceConfiguration xsuaaConfig, OAuth2ServiceConfiguration oauthConfig) {
```

Comment:
Method is no specific to this test and therefore should be extracted into a separate class (see also other review comment)